lib/jobs/install-os.js

```diff
@@ -167,32 +167,45 @@ function installOsJobFactory(
         var self = this;
         var disk = self.options.installDisk;
 
-        //If disk is empty then do nothing.
         //If disk is string, it means user directly input the drive wwid, so don't need conversion.
-        if ((!disk && disk !== 0) || _.isString(disk)) {
+        if (_.isString(disk) && !_.isEmpty(disk)) {
             return Promise.resolve(disk);
         }
 
-        if (!_.isNumber(disk)) {
+        if (!_.isNumber(disk) && !_.isEmpty(disk)) {
```

Comment:
it's better to have a comment here why the disk can be empty, if people is not familiar with our design, he may feel a little confused